rented-properties/src/main/java/org/egov/cpt/service/AccountStatementExcelGenerationService.java

```diff
@@ -0,0 +1,166 @@
+				+ "% P.A as per clause 15 of Lease Deed");
+		try {
+			Workbook workbook = new XSSFWorkbook();
+			Sheet sheet = workbook.createSheet("AccountStatement");
+
+			Font headerFont = workbook.createFont();
+			headerFont.setBold(true);
+			headerFont.setFontHeightInPoints((short) 10);
+			headerFont.setColor(IndexedColors.BLACK.getIndex());
+
+			// Create a CellStyle with the font
+			CellStyle headerCellStyle = workbook.createCellStyle();
+			headerCellStyle.setFont(headerFont);
+
+			// Create a Row
+			Row headerRow = sheet.createRow(0);
+
+			Cell cell = headerRow.createCell(0);
+			cell.setCellValue(" ");
+			cell.setCellStyle(headerCellStyle);
+
+			cell = headerRow.createCell(1);
+			cell.setCellValue("Provisional Statement of Plot No. ,");
+			cell.setCellStyle(headerCellStyle);
+
+			Row headerRow2 = sheet.createRow(1);
+			cell = headerRow2.createCell(0);
+			cell.setCellValue(" ");
+			cell.setCellStyle(headerCellStyle);
+
+			cell = headerRow2.createCell(1);
+			cell.setCellValue("Vikas Nagar, Mauli Jagaran, UT Chandigarh");
+			cell.setCellStyle(headerCellStyle);
+			int j = 0;
+			for (int i = 3; i < 12; i++) {
+				Row headerRow3 = sheet.createRow(i);
+				cell = headerRow3.createCell(0);
+				if (j < propertyColumns.length) {
+					cell.setCellValue(propertyColumns[j]);
+					cell.setCellStyle(headerCellStyle);
+				}
+				cell = headerRow3.createCell(1);
+				if (j < propertyList.size()) {
+					cell.setCellValue(propertyList.get(j));
+				}
+				j++;
+			}
+
+			Row headerRow1 = sheet.createRow(13);
+			for (int i = 0; i < columns.length; i++) {
+				cell = headerRow1.createCell(i);
+				cell.setCellValue(columns[i]);
+				cell.setCellStyle(headerCellStyle);
+			}
+
+			int rowNum = 14;
+			for (RentAccountStatement rentAccountStmt : accountStatementResponse.getRentAccountStatements()) {
+				Row row = sheet.createRow(rowNum++);
+
+				row.createCell(0).setCellValue(Instant.ofEpochMilli(rentAccountStmt.getDate())
+						.atZone(ZoneId.systemDefault()).toLocalDate().format(FORMATTER));
+				row.createCell(1).setCellValue(rentAccountStmt.getAmount());
+				row.createCell(2).setCellValue(rentAccountStmt.getType().toString());
+				row.createCell(3).setCellValue(rentAccountStmt.getRemainingInterest());
+				row.createCell(4).setCellValue(rentAccountStmt.getRemainingPrincipal());
+				row.createCell(5).setCellValue(rentAccountStmt.getDueAmount());
+				row.createCell(6).setCellValue(rentAccountStmt.getRemainingBalance());
+			}
+
+			System.out.println(sheet);
+			System.out.println(headerRow);
+			// Write the output to a file
+			File file = new File("poi-generated-file.xlsx");
+			FileOutputStream fileOut = new FileOutputStream(file);
+
+			String response = fileStoreUtils.fetchFileStoreId(file, properties.get(0));
+
+			workbook.write(fileOut);
+			fileOut.close();
+
+			// Closing the workbook
+			workbook.close();
+			return response;
+		} catch (Exception e) {
+			log.error(e.getMessage());
+		}
+		return null;
```

Comment:
Instead of returning `null` on failure, throw an exception. `throw new CustomException("XLS_NOT_GENERATED", "Could not generate account statement");`

Reply:
exception thrown as recommended